e2e/ansible/playbooks/hyperconverged/test-k8s-jenkins-upgrade/k8s-jenkins-upgrade.yml

```diff
@@ -0,0 +1,259 @@
+#8.Perform build operation with the existing created job.
+#9.Compare both the jobs buil.
+#(Way to Check for data persitency inside application pod.)
+
+###############################################################################################
+
+- hosts: localhost
+
+  vars_files:
+    - k8s-jenkins-pod-vars.yml
+ 
+  tasks:
+
+   - block:
+
+     - include: k8s-jenkins-pod-prereq.yml
+
+     - name: 1)  Check whether maya-apiserver is deployed
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+       vars:
+          ns: "{{ operator_ns }}"
+          app: apiserver
+
+      
+     - name: 2)  Download YAML for jenkins 
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/get_url.yml"
+       vars:
+         url: "{{ jenkins_link }}"
+         dest: "{{ result_kube_home.stdout }}/{{ pod_yaml_alias }}"
+
+
+     - name: 2a) Replace jenkins pod version
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/regex_task.yml"
+       vars:
+         path: "{{ result_kube_home.stdout }}/jenkins.yml"
+         regex1: "{{replace_item}}"
+         regex2: "{{replace_with}}"
+       delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+    
+     - name: Create namespace for deployment.
+       shell: source ~/.profile; kubectl create ns {{namespace}}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+      
+     - name: 3) Deploy jenkins pod 
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_task.yml"
+       vars:
+          app_yml: "{{ pod_yaml_alias }}"
+          ns: "{{ namespace }}"
+    
+     - name: 3a) Confirm pod status is running
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+       vars:
+          ns: "{{ namespace }}"
+          app: jenkins
+
+     - name: 3b) Verify that the jenkins is a cluster service
+       shell: source ~/.profile; kubectl get svc -n {{namespace}}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: result_service
+       failed_when: "'jenkins-svc' not in result_service.stdout"
+
+     - name: 3c) Get the name of the pod.
+       shell: source ~/.profile; kubectl get pods -n {{namespace}} | grep jenkins | awk {'print $1'}
+       args: 
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: jenkins_pod_name
+
+         
+     - name: 4) Download Jenkins-cli.jar file inside /var/jenkins_home of pod.
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{namespace}} 
+          -- wget -P /var/jenkins_home http://localhost:8080/jnlpJars/jenkins-cli.jar
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       until: "'saved' in result.stdout"
+       delay: 30
```

Comment:
@vibhor995 Can you explain how are you validating this task? because no register variable is used but you are grepping for saved string in result.stdout.